### PR TITLE
Add end user documentation to cilium

### PR DESCRIPTION
**Pull request by DockToFuture**
**What this PR does / why we need it**:
Add documentation for end user to cilium

**Release note**:

```improvement operator
None
```

### CODE REVIEW

File: docs/usage-as-end-user.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+The `hubble.enabled` field describes whether hubble should be deployed into the cluster or not.
+
+The `debug` field describes whether you want to run cilium in debug mode or not, change this value to true to use debug mode.
+
+The `psp` field describes whether `cilium-operator` and `cilium-agent` shall be deployed with pod security policies or not.
+
+The `tunnel` field describes the encapsulation mode for communication between nodes. Possible values are `vxlan`, `geneve` or `disabled`.
+
+The `store` field describes which backend to use to store the identities. Can be either `etcd` (kvstore) or `kubernetes` (crd).
```
Comment (vpnachev) [inline docs/usage-as-end-user.md:26]: Mention which are the default values.

File: docs/usage-as-end-user.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+
+## Example `NetworkingConfig` manifest
+
+An example `NetworkingConfig` for the Cilium extension looks as follows:
+
+```yaml
+apiVersion: cilium.networking.extensions.gardener.cloud/v1alpha1
```
Comment (vpnachev) [inline docs/usage-as-end-user.md:34]:
```suggestion
apiVersion: cilium.networking.extensions.gardener.cloud/v1alpha1
kind: NetworkConfig
```

File: docs/usage-as-end-user.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+hubble:
+  enabled: true
+#debug: "false"
+#psp: true
+#tunnel: vxlan
+#store: kubernetes
+kind: NetworkConfig
```
Comment (vpnachev) [inline docs/usage-as-end-user.md:41]:
```suggestion
```

File: docs/usage-as-end-user.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+```
+
+## Example `Shoot` manifest
+
+Please find below an example `Shoot` manifest with cilium networking configuration:
+
+```yaml
+kind: Shoot
+apiVersion: core.gardener.cloud/v1beta1
```
Comment (vpnachev) [inline docs/usage-as-end-user.md:50]:
```suggestion
apiVersion: core.gardener.cloud/v1beta1
kind: Shoot
```

**Review by vpnachev — DISMISSED** [review]
Thanks!

Looks very well, I have just a few minor nits.

File: docs/usage-as-end-user.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+```yaml
+apiVersion: cilium.networking.extensions.gardener.cloud/v1alpha1
+kind: NetworkConfig
+hubble:
+  enabled: true
+#debug: false
+#psp: true
+#tunnel: vxlan
+#store: kubernetes
+```
```
Comment (rfranzke) [inline docs/usage-as-end-user.md:41]: I would move this example up to line ~18 (before the detailed explanation of the various fields follows)

File: docs/usage-as-end-user.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    timeWindow:
+      begin: 000000+0000
+      end: 010000+0000
+    autoUpdate:
+      kubernetesVersion: true
+      machineImageVersion: true
+  hibernation:
+    schedules:
+      - start: '00 17 * * 1,2,3,4,5'
+        location: Europe/Berlin
```
Comment (rfranzke) [inline docs/usage-as-end-user.md:116]: Hm, generally, this is hard to maintain for a non-provider extension. What about only proposing the `networking` fields?

```suggestion
  networking:
    type: cilium
    providerConfig:
      apiVersion: cilium.networking.extensions.gardener.cloud/v1alpha1
      kind: NetworkConfig
      hubble:
        enabled: true
    pods: 100.96.0.0/11
    nodes: 10.250.0.0/16
    services: 100.64.0.0/13
  ...
```

Comment (vpnachev) [reply docs/usage-as-end-user.md:116]: If the provider configs are skipped here, I would add something like
> If you would like to see provider specific shoot example, please check out the documentation of the well know extensions, list of which can be found [here](https://github.com/gardener/gardener/tree/master/extensions#infrastructure-provider).

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**